dist_run.py

```diff
@@ -122,6 +112,88 @@ def _load_model_weights(stage_module, hf_model_name, device, model_config):
         raise ValueError(f"Missing {num_missing_weights} weights")
 
 
+def _encode_strings(
+    strings: List[str],
+    tokenizer,
+    bos: bool = True,
```

Comment:
nit: what does `bos` mean?

Reply:
bos = Beginning of Sequence. 
It's the prompt used to start off an interaction, visually it's "<|startoftext|>". 
We put that as the first token for a new prompt submission.